### ISSUE TITLE
Missing FROM clause when where + has_one

### ISSUE THREAD

**Issue by justincampbell** [Casecommons/pg_search#246]
I'm trying to setup pg_search using `pg_search_scope`. The model that I'm searching has a few has_one relations. These work fine with `associated_against`, but I can't figure out how to add a where clause. In my use case, I want to filter `where(other: { is_private: false })`, or `where(other: { model: "foobar" })`.

```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "others"
```

I've also tried a few permutations of the key/table name, adding `references`, `includes`, and `joins`.

I have a failing test case in https://github.com/justincampbell/pg_search/tree/missing-from-clause-entry

Is there a way to do this with `pg_search_scope`? Am I doing something wrong? Any help would be appreciated. Thanks!

**Comment by justincampbell** [comment]
btw I'm on Rails 4.1, PostgreSQL 9.4.1

**Comment by DmitryKK** [comment]
+1

**Comment by nertzy** [comment]
Does the same query work without the search scope chained on? I would start there. 

If you're on pg_search 1.0.x, then the associated_against part of the query should only affect the pg_search subquery, which I think won't affect any of the other scopes you've chained on, but I could be mistaken.

**Comment by justincampbell** [comment]
@nertzy Yes, removing the `.named()` in this test does not result in an error: https://github.com/justincampbell/pg_search/commit/f42cdcd996a42c40f9e2edcc251272ac7e2ebb8c
